### ISSUE TITLE
Powermapper flags WCAG 2.0 F24 failure for the textarea component

### ISSUE THREAD

**Issue by 36degrees** [alphagov/govuk-frontend#1772]
As reported on Slack:

> Hello, we’re using an automated tool to check our design system built pages for accessibility problems and we get an error on a text-area: 
>
> If you set any of the colors on the body or a elements you must set all of them. 
> WCAG 2.0 AA F24Section 508 (2017) AA F24

Which is [F24: Failure of Success Criterion 1.4.3, 1.4.6 and 1.4.8 due to specifying foreground colors without specifying background colors or vice versa](https://www.w3.org/TR/WCAG20-TECHS/F24.html)

The automated tool that flagged this is [Powermapper](https://www.powermapper.com/products/sortsite/).

**Comment by 36degrees** [comment]
As far as I can see neither the background colour nor the foreground colour are set on textareas, so I’m not sure why it’s complaining…

We should be careful if we decide to change the background / foreground colours of inputs as it's caused problems in the past – see https://accessibility.blog.gov.uk/2017/03/27/how-users-change-colours-on-websites and https://github.com/alphagov/govuk_elements/pull/397.

**Comment by 36degrees** [comment]
The procedure for testing F24 only talks about foreground and background colours, neither of which are set on the textarea component.

The only colour that we do set is the border colour, and [technique C25](https://www.w3.org/TR/WCAG20-TECHS/C25.html) suggests that this is a totally fine thing to do.

I _think_ this might be a bug in Powermapper's implementation of the F24 test?

**Comment by 36degrees** [comment]
I've not yet been able to reproduce this using Powermapper's online demo against http://govuk-frontend-review.herokuapp.com/components/textarea/preview.

(I had to run it locally and remove the robots.txt restriction to get it to run)

I've asked the person that originally raised this if they have any additional styles that might be setting the background or foreground colour on the textarea.

It could be that the desktop version behaves differently, or they're using an older version?

**Comment by 36degrees** [comment]
@antimega has been able to reproduce this in SortSite:

![Screenshot 2020-03-30 at 14 39 39](https://user-images.githubusercontent.com/121939/78006152-8714c380-7334-11ea-924c-45a073eb239b.png)

> If you set any of the colors on the body or a elements you must set all of them.	WCAG 2.1 AA F24	1 pages
>
> In HTML the color attributes are text, bgcolor, link, alink and vlink. In CSS the attributes are color and background-color. Some users have browser defaults set to white text on a black background, so setting one color without setting the others can result in black text on a black background.		
> 
>The colors used are:
color: rgb(0,0,0); uses browser default background-color: rgb(255,255,255);
>
> http://govuk-frontend-review.herokuapp.com/components/textarea/preview	Line 29

Line 29 is the `<body>` tag, which does set the background colour but not the foreground colour (which is the exact opposite of what the SiteSort error says).

We're also seeing this error flagged on http://govuk-frontend-review.herokuapp.com/components/back-link/preview, so it looks like this might not be related to the textarea component at all.

**Comment by nikola-naydenov-hmcts** [comment]
Hi, we are getting the same type of issues from our accessibility tests. We use pa11y.
![image](https://github.com/alphagov/govuk-frontend/assets/47384516/5fb34d8e-20c6-44ca-9310-a4559ad7a753)

It's actually complaining that the foreground colour needs to have a contrast of 7:1 with the white background colour.
What would you suggest?

**Comment by querkmachine** [comment]
Hi @nikola-naydenov-hmcts,

The tool there appears to be testing against WCAG Level AAA, and is reporting that blue links don't have sufficient contrast with the white background.

Public services in the UK are expected to comply with Level AA which only requires a contrast ratio of 4.5:1, and GOV.UK frontend is built to match that.

If you are intending to meet Level AAA, you'll need to customise Frontend to match those requirements. Otherwise, you may have to change the settings on the auditing tool to ignore Level AAA failures. 

Hopefully that helps!
